src/agents/regulator.jl

```diff
@@ -999,28 +1009,30 @@ function solve_agent_problem!(
                 ) * (1 - der_aggregator.aggregation_level(reg_year_index_dera, z))
                 for d in model_data.index_d, t in model_data.index_t
             ) -
-            # green technology subscription
+
             sum(
-                model_data.omega(d) * delta_t * utility.rho_C_my(j, z, d, t) * sum(green_developer.green_tech_buildout_my(Symbol(Int(y_symbol)), j, z, h) for y_symbol in
-                model_data.year(first(model_data.index_y_fix)):reg_year)
+                model_data.omega(d) * delta_t * utility.rho_C_my(j, z, d, t) *
+                sum(
+                    green_developer.green_tech_buildout_my(Symbol(Int(y_symbol)), j, z, h)
+                    for y_symbol in model_data.year(first(model_data.index_y_fix)):reg_year
+                )
                 for j in model_data.index_j, d in model_data.index_d, t in model_data.index_t
             )
     end
 
     net_demand_wo_green_tech_h_wo_loss = make_keyed_array(model_data.index_z, model_data.index_h)
-    for z in model_data.index_z, h in model_data.index_h
+    for (z, h) in model_data.index_z_h_map
         net_demand_wo_green_tech_h_wo_loss(z, h, :) .= 
             # Demand with loss
             sum(
                 customers.gamma(z, h) * model_data.omega(d) * delta_t * customers.d(h, z, d, t) / (1 + utility.loss_dist) for
                 d in model_data.index_d, t in model_data.index_t
             ) -
             # DG
-            # since this net demand is for rate calculation, we need to consider energy offset at the household level.
-            # e.g. two household, with 100 MW load each (without loss), if one of them has DER and generated 
-            # 120 MW of energy, he/she does not need to pay for energy, but the other one still have to pay 
-            # 100 MW instead of 80 MW.
-            # behind the meter generation for pv-only customers
+            # Since this net demand is for rate calculation, we need to consider energy offset at the household level.
+            # For example, two households with 100 MW load each (without loss); if one has DER generating
+            # 120 MW of energy, they do not need to pay for energy, but the other still has to pay 100 MW instead of 80 MW.
+            # Behind-the-meter generation for PV-only customers
```

Comment:
Similar text was deleted above. Was that on purpose? (Is this text more relevant here than above where it was deleted?)